include/event2/event_watcher.h

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2000-2007 Niels Provos <provos@citi.umich.edu>
+ * Copyright (c) 2007-2012 Niels Provos and Nick Mathewson
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the author may not be used to endorse or promote products
+ *    derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef EVENT2_EVENT_WATCHER_H_INCLUDED_
+#define EVENT2_EVENT_WATCHER_H_INCLUDED_
+
+/** @file event2/event_watcher.h
+
+	"Prepare" and "check" watchers. A "prepare" watcher is a callback that fires immediately before
+	polling for I/O. A "check" watcher is a callback that fires immediately after polling and before
+	processing any active events. This may be useful for embedding other libraries' event loops (e.g.
+	UI toolkits) into libevent's. The timing information exposed by the callbacks is especially handy
+	for server health and performance monitoring.
+
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef EVENT__HAVE_SYS_TIME_H
+#include <sys/time.h>
+#endif
+#include <time.h>
+
+#include <event2/visibility.h>
+
+struct event_base;
+struct event_watcher;
+
+/**
+	Contextual information passed from event_base_loop to the "prepare" watcher callbacks. We define
+	this as a struct rather than individual parameters to the callback function for the sake of future
+	extensibility.
+ */
+struct prepare_watcher_cb_info {
```

Comment:
- this does not contains `event_` prefix (and not only this one)
- but please do not call it `event_prepare_watcher_cb_info`, tooooo long
- and if this `struct` exposed then you will not have ability to change it, maybe it is better to hide it in module part and provide getter for the only field that it has, in this case user will not allow to change our timeout (since it has a pointer to our internal variable) which is not good from some point of vew